src/leap/mail/mail.py

```diff
@@ -837,10 +854,16 @@ def get_collection_by_mailbox(self, name):
         """
         :rtype: MessageCollection
         """
+        collection = self._collection_mapping.get(name, None)
+        if collection:
+            return collection
```

Comment:
So, this method is actually returning either a MessagecCollection or a Deferred that will fire with a MessageCollection. I'm not sure if this is indeed the expected behaviour and documentation, but if not maybe we could always return a Deferred, maybe like this:

```python
    if collection:
        return defer.Deferred().addCallback(lambda _: collection)
```

Or something like this.